Dockerfile.client.prod

```diff
@@ -0,0 +1,38 @@
+# Build all the things
+FROM node:18-alpine AS build
```

Comment:
Use node:20.4-bullseye-slim instead, unless there's a good reason not to
- latest nodejs lts version
- debian based, less pain
- os versions pinned (sort of) so you avoid us-in-6-months deploy issues

Reply:
Also: could these be moved inside their respective cms/client folders?

Reply:
Also: do we really need separate docker files for dev and prod? I can see this growing for testing, and for staging, and for whetever else, all to save some lousy hundred MBs on a hard drive somewhere that costs next to nothing.

Reply:
the difference is in the NODE_ENV and in using yarn start vs yarn dev. I don't know how to improve on the duplication of dev / prod, but I think we won't need a separate one for staging / prod.

Reply:
I though there is some reason why @davidsingal used node version 18, but I can give it try with latest version and see if something blows up or not. 

Agree that It makes more sense to move them to respective folders, will try to do it --> need to just rewrite paths in dockers and docker-compose. 

I prefer to keep prod and dev dockerfiles separate --> never needed dockerfiles for test or staging... Basically main difference is that you don't need a lot extra steps for dev docker image so building time is a lot faster, but image is often bigger. Also a lot of stuff can be done on the fly with dev image, but you wanna have everything precompiled with Stage/Prod. It is often worthy to do couple extra steps at prod dockerfiles which makes image more optimal. Smaller images are also quicker to transfer and faster to deploy so there are other advantages, not just hard drive price....

Reply:
I don't understand the case where a dev Docker image is needed, but not a test one. If you have a docker image (like this one) where running "yarn dev" is hardcoded, I assume you are developing locally using Docker, in which case you'll also need a different Docker file that runs "yarn test" to run the tests.

The problem with this approach is that now you have 3 Docker files (per project, so 6 in total) to maintain. And every time a change pops up, you have to do it and test it 3 times. And I have a bunch of projects that prove that it won't happen.

Reply:
This can be easily fixed by entrypoint which you have mentioned above -- don't mind to add it ;). Even with entrypoint and running different command based on ENV, it still makes sense to keep two different dockerfiles from my point of view, which generate less or more optimal image. On other hand I agree that It is extra stuff to maintain -- luckily building process does not change every day...

Reply:
Up to you on how you want to proceed - I see the point of having a prod vs local image, I just don't think the negligible impact the docker image size has is worth the tradeoff of having another docker file to maintain. but, like I said, up to you, it's a subjective matter

Reply:
Tried to take into account all your notes @tiagojsag with only one exception:
 - debian based images --> hopefully it is worthy, because it takes more time to build and images are bigger :)
 - os versions pinned
 - moved dockefiles to their respective folders
 - added entrypoint so you can easily run dev/test via docker-compose
Dockerfiles are using `node:18.15.0-bullseye-slim`, which is because version of node which project uses now. I don't think it is difficult to upgrade this at future, but it is outside of my expertise to switch whole project to different node version...